cruise-control/src/test/java/com/linkedin/kafka/cruisecontrol/monitor/sampling/aggregator/KafkaPartitionMetricSampleAggregatorTest.java

```diff
@@ -257,7 +257,7 @@ public void testTooManyFlaws() throws NotEnoughValidWindowsException {
           metricSampleAggregator.aggregate(clusterAndGeneration(metadata.fetch()),
                                            NUM_WINDOWS * WINDOW_MS,
                                            new OperationProgress());
-      assertTrue(result.valuesAndExtrapolations().isEmpty());
+      assertEquals(NUM_WINDOWS - 3, result.valuesAndExtrapolations().get(PE).windows().size(), 0);
```

Comment:
Why are we using the `float` comparator for the `assert`s added in this patch?

Reply:
Changed to `int` comparator.

Reply:
I added a comment about what we expect in this test.  The changes in expectation is because window with zero valid partition sample is no longer taken as valid window(window 1,2,3 in this test). And valid partition in aggregate result is partition which has valid sample in all valid windows( an intersection of valid partition set from all valid partition).  After the exclusion of window 1,2,3, partition `topic-0` now is a valid partition in aggregate result.